File: Api/src/test/java/net/binggl/mydms/test/documents/DocumentModelTest.java
package net.binggl.mydms.test.documents;

import static io.dropwizard.testing.FixtureHelpers.fixture;
import static org.assertj.core.api.Assertions.assertThat;

import java.util.Date;
import java.util.UUID;

import org.joda.time.DateTime;
import org.joda.time.DateTimeZone;
import org.junit.Test;

import com.fasterxml.jackson.databind.ObjectMapper;

import io.dropwizard.jackson.Jackson;
import net.binggl.mydms.features.documents.Document;

public class DocumentModelTest {

	private static final ObjectMapper MAPPER = Jackson.newObjectMapper();
	
    @Test
    public void serializesAndDeserializeJSON() throws Exception {
        final DateTime stamp = new DateTime(2016,10,1,0,0,1,1, DateTimeZone.UTC);
        final Date date = stamp.toDate();
        
    	final Document document = new Document(UUID.fromString("a08bee51-3b09-4761-bc93-6f2e10b4f366"), "document", "filename", "alternativeId", "previewLink", 0.0, date, date);

        final String expected = MAPPER.writeValueAsString(
                MAPPER.readValue(fixture("fixtures/document.json"), Document.class));
        assertThat(MAPPER.writeValueAsString(document)).isEqualTo(expected);
    }
}

File: Api/src/test/kotlin/net/binggl/mydms/features/records/DocumentModelTest.kt
package net.binggl.mydms.features.records

import FixtureHelpers
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import net.binggl.mydms.features.records.model.Document
import org.junit.Assert
import org.junit.Test
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class DocumentModelTest {

    @Test
    fun serializeToJson(): Unit {
        val expected = MAPPER.writeValueAsString(MAPPER.readValue(FixtureHelpers.fixture("fixtures/document.json"), Document::class.java))
        Assert.assertEquals(expected, MAPPER.writeValueAsString(document))
    }

    @Test
    fun deserializeFromJson(): Unit {
        Assert.assertEquals(MAPPER.readValue(FixtureHelpers.fixture("fixtures/document.json"), Document::class.java), document)
    }

    private val document: Document
        get() {
            return Document(id = "id", title = "document", fileName = "filename", alternativeId = "alternativeId",
                    previewLink = "previewLink", amount = 0.0, created = LocalDateTime.parse("2018-01-01T00:00:00.000", fmt),
                    modified = LocalDateTime.parse("2018-01-01T00:00:00.000", fmt), tags = emptyList(),
                    senders = emptyList(), uploadFileToken = null)
        }

    companion object {
        // https://github.com/FasterXML/jackson-module-kotlin
        private val MAPPER = jacksonObjectMapper().registerModule(JavaTimeModule())
        private val fmt = DateTimeFormatter.ISO_DATE_TIME
    }
}